python/common/org/python/types/Int.java

```diff
@@ -78,7 +78,15 @@ public Int(org.python.Object[] args, java.util.Map<java.lang.String, org.python.
                     );
                 }
             }
-        } else if (args.length > 3) {
+        } else if (args.length == 2) {
+            try {
+                org.python.types.Str argStr = (org.python.types.Str) args[0];
+                org.python.types.Int argInt = (org.python.types.Int) args[1];
+                this.value = java.lang.Integer.parseInt(argStr.value, (int) argInt.value);
+            } catch (Exception e) {
+                throw new org.python.exceptions.ValueError("invalid literal for int() with base " + args[1] + ": '" + args[0] + "'");
+            }
+        } else if (args.length > 2) {
             throw new org.python.exceptions.NotImplementedError("int() with a base is not implemented");
```

Comment:
This `NotImplementedError` message doesn't make sense, we should throw a different error message here, as CPython does:

```
>>> int('a', 16, 3)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: int() takes at most 2 arguments (3 given)
```